ShotgunImportExport/Sources/CSVImportExport.cs

```diff
@@ -0,0 +1,123 @@
+    static private readonly string k_pathToFile = Path.Combine("Assets", "CsvImportExport", "Editor");
+    static private readonly string k_timelineObjectName = "Sample Shotgun CSV timeline";
+
+    private static Tuple<ActivationTrack, TimelineClip> CreateActivationTrackAndClip(TimelineAsset timeline, PlayableDirector director)
+    {
+            ActivationTrack track = timeline.CreateTrack<ActivationTrack>(null, "");
+            TimelineClip clip = track.CreateDefaultClip();
+            return new Tuple<ActivationTrack, TimelineClip>(track, clip);
+    }
+
+    [UnityEditor.MenuItem("Shotgun CSV/Import")]
+    public static void ImportCsv ()
+    {
+        TimelineAsset timeline;
+        PlayableDirector director;
+
+        //Create a Director object and a timeline
+        UnityEngine.GameObject director_object = new UnityEngine.GameObject();
+        director = (PlayableDirector)director_object.AddComponent(typeof(PlayableDirector));
+        director.name = k_directorObjectName;
+        timeline = ScriptableObject.CreateInstance("TimelineAsset") as TimelineAsset;
+        timeline.name = k_timelineObjectName;
+        director.playableAsset = timeline;
+        
+        foreach (var row in ReadCsvFile(Path.Combine(k_pathToFile, k_fileName)))
+        {
+            var CsvField = CreateActivationTrackAndClip(timeline, director);
+            CsvField.Item1.name = row["Shot Code"];
+            CsvField.Item2.start =  (Convert.ToDouble(row["Cut In"]) / k_FPS);
+            CsvField.Item2.duration = ((Convert.ToDouble(row["Cut Out"]) - Convert.ToDouble(row["Cut In"])) / k_FPS);
+        }
+    }
+
+    [UnityEditor.MenuItem("Shotgun CSV/Export")]
+    public static void ExportCsv()
+    {
+        GameObject director = GameObject.Find(k_directorObjectName);
+
+        // Get the actual timeline object
+        var timelineAsset = (director.GetComponent("PlayableDirector") as PlayableDirector).playableAsset as TimelineAsset;
+
+        // Each dictionary in the list represents a row in the CSV file
+        // Each value in the row is addressable by its column name
+        var toWrite = new List<Dictionary<string, string>>();
+
+        foreach (var track in timelineAsset.GetOutputTracks())
+        {
+            var dict = new Dictionary<string, string>();
+            var start = Convert.ToInt64(track.start * k_FPS);
+            var end = Convert.ToInt64((track.duration + track.start) * k_FPS);
+            var shotCode = track.name;
+
+            dict.Add("Shot Code", shotCode);
+            dict.Add("Cut In", start.ToString());
+            dict.Add("Cut Out", end.ToString());
+            toWrite.Add(dict);
+        }
+
+        var pathToOutputFile = Path.Combine(k_pathToFile, "export_from_unity.csv");
+        Debug.Log(pathToOutputFile);
+        WriteCsv(toWrite, pathToOutputFile);
+    }
+
+    private static List<Dictionary<string, string>> ReadCsvFile(string filePath)

```

Comment:
Parsing a CSV would normally give you a list of rows (list of strings). Why create a list of dictionaries where each dict's keys are the same?

Reply:
we decided it doesn't matter for sample code for small files